File: src/main/java/core/listeners/ReactionListener.java
package core.listeners;

import modules.poll.SmashPassModule;
import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import utils.BotUtils;


public class ReactionListener extends ListenerAdapter {

    @Override
    public void onMessageReactionAdd(MessageReactionAddEvent event) {
        if (event.getUser() == BotUtils.bot.getSelfUser()) {
            return;
        }
        if (BotUtils.smashPassInstances.containsKey(event.getMessageId())) {
            SmashPassModule.addReaction(event);
        }
    }

    @Override
    public void onMessageReactionRemove(MessageReactionRemoveEvent event) {
        if (event.getUser() == BotUtils.bot.getSelfUser()) {
            return;
        }
        if (BotUtils.smashPassInstances.containsKey(event.getMessageId())) {
            SmashPassModule.removeReaction(event);
        }
    }
}

File: src/main/kotlin/com/github/sylux6/watanabot/core/listeners/ReactionListener.kt
package com.github.sylux6.watanabot.core.listeners

import com.github.sylux6.watanabot.utils.bot.bot
import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent
import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class ReactionListener : ListenerAdapter() {

    override fun onMessageReactionAdd(event: MessageReactionAddEvent) {
        if (event.user === bot.selfUser) {
            return
        }
//        if (BotUtils.smashPassInstances.containsKey(event.messageId)) {
//            SmashPassModule.addReaction(event)
//        }
    }

    override fun onMessageReactionRemove(event: MessageReactionRemoveEvent) {
        if (event.user === bot.selfUser) {
            return
        }
//        if (BotUtils.smashPassInstances.containsKey(event.messageId)) {
//            SmashPassModule.removeReaction(event)
//        }
    }
}
